### PR TITLE
[ENG-2656, WIP] Adds workflow history route, store

### PR DESCRIPTION
**Pull request by vsreekanti**
## Describe your changes and why you are making these changes

This PR adds an API route that returns the version history (version ID, timestamp, status) for a particular workflow. This route is hooked up to the version selector on the frontend. This should also be used for the workflow header soon, but I wanted to keep this PR tactical to start since we aren't deprecating any code yet.

This is an example implementation of the types of API routes that I'd like to see us moving towards in the future. I implemented this in about ~2.5 hours including debugging*, which gives me reasonable confidence that these smaller routes are both easier to implement and easier to maintain. 

The full set of new code that's added is ~250 LOC, which includes a bunch of boilerplate, and I think we'll be able to deprecate a lot of the really complex reducers that we have in the codebase right now.

*To be fair, I haven't added any tests.

## Related issue number (if any)

ENG-2656

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/handler/get_workflow_history.go
Comment (kenxu95) [inline src/golang/cmd/server/handler/get_workflow_history.go:32]: Since we're documenting the request, it might be good to document our responses too. This route is fairly straightforward but we should definitely cover assumptions like what is the ordering of the versions in the list.

**Review by kenxu95 — COMMENTED** [review]
Going to leave UI review to Andre. But having a historical version endpoint makes a ton of sense to me! I imagine we'll have a similar one for artifacts too?

File: src/ui/common/src/components/workflows/version_selector.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/version_selector.tsx:37]: nit: `useState<number>(0)`

File: src/ui/common/src/components/workflows/version_selector.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/version_selector.tsx:61]: do we need to add null checking for exec_state here? I'm not entirely sure just by looking at the code.

File: src/ui/common/src/reducers/workflowHistory.ts
Comment (agiron123) [inline src/ui/common/src/reducers/workflowHistory.ts:7]: File a linear task here?

**Review by agiron123 — APPROVED** [review]
UI Code looks good here and follows the handlers pattern that we're trying to move our requests to.

Left some minor comments, but should be good to go after resolving.

**Review by likawind — APPROVED** [review]
@vsreekanti is this for showing the version selector in workflow headers? (nvm, saw this in your PR comment)

**Comment by likawind** [comment]
Another suggestion is to look at this file: https://github.com/aqueducthq/aqueduct/blob/main/src/golang/lib/workflow/dag/response.go there may be a few structs we can reuse here, but not a blocker for this PR

**Comment by vsreekanti** [comment]
Addressed the feedback here. Just waiting for confirmation on the DB guarantees and then will merge.

File: src/ui/common/src/components/workflows/version_selector.tsx
Comment (vsreekanti) [reply src/ui/common/src/components/workflows/version_selector.tsx:61]: Confirmed with Saurav that this is correct.

**Pull request merged**
